script.js

```diff
@@ -1,4 +1,668 @@
+// Game modes stored in constants
+const WAITING_FOR_NUM_PLAYERS = "waiting for num of players";
+const WAITING_FOR_NAME = "waiting for username";
+const INSTRUCTIONS = "tell player instructions";
+const ASK_FOR_BET = "ask player for betting amount";
+const TAKE_BET = "take user's bet";
+const DEAL_CARDS = "deal cards";
+const SELECT_PLAYER = "select player and check for blackjack";
+const PLAYER_ACTION = "player hits or stands";
+const CHANGE_ACE_MODE = "change ace mode";
+const COMPUTER_CALCULATES = "computer calculates";
+const FINAL_RESULT = "final result";
+const GAME_SUMMARY = "output all results";
+
+// Declare user inputs as constants
+const INPUT_HIT = "hit";
+const INPUT_STAND = "stand";
+const INPUT_CHANGE_ACE = "ace";
+const POSITION = "position";
+const ACE_ONE = "ace 1";
+const ACE_ELEVEN = "ace 11";
+const INPUT_RESET = "reset";
+
+// Global variables
+var allPlayerOutcomes = [];
+var allPlayerCards = [];
```

Comment:
since you are using const for your variables whose value does not change, you should use let for variables whose value will change instead. This is because var doesn't tell me if the variable value will change or not. So instead of var allPlayerOutcomes, it would be clearer if written as let allPlayerOutcomes. Let behaves like var, except it signals that this variable's value will be updated further down in the code.